actions/setup-gap/action.yml

```
@@ -79,31 +106,128 @@ runs:
+
+        # Remove certificate-authority-data populated from `aws eks update-kubeconfig` which is for the endpoint which is not publicly accessible
+        kubectl config unset clusters.$CLUSTER_ARN.certificate-authority-data
+
+    - name: Login to AWS ECR (public ecr only)
+      if: inputs.use-private-ecr-registry != 'true'
       uses: aws-actions/amazon-ecr-login@062b18b96a7aff071d4dc91bc00c4c1a7945b076 # v2.0.1
       env:
         AWS_REGION: us-east-1
       with:
         registry-type: public
 
-    - name: Run aws sig4 proxy container
+    - name: Run aws sig4 proxy container (public ecr only)
+      if: inputs.use-private-ecr-registry != 'true'
       shell: bash
       env:
         AWS_ACCESS_KEY_ID: ${{ env.AWS_ACCESS_KEY_ID }}
         AWS_SECRET_ACCESS_KEY: ${{ env.AWS_SECRET_ACCESS_KEY }}
         AWS_SESSION_TOKEN: ${{ env.AWS_SESSION_TOKEN }}
       run: |
         docker run --rm -d \
-          -e "AWS_ACCESS_KEY_ID=${AWS_ACCESS_KEY_ID}" \
-          -e "AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY}" \
-          -e "AWS_SESSION_TOKEN=${AWS_SESSION_TOKEN}" \
+          -e AWS_ACCESS_KEY_ID \
+          -e AWS_SECRET_ACCESS_KEY \
+          -e AWS_SESSION_TOKEN \
           -p "${{ inputs.proxy-port }}":"${{ inputs.proxy-port }}" \
           public.ecr.aws/aws-observability/aws-sigv4-proxy:${{ inputs.proxy-version }} \
           --name execute-api --region ${{ inputs.aws-region }} \
           --host "${{ inputs.api-gateway-host }}" \
           --log-failed-requests
 
+    - name: Login to AWS ECR (private ecr only)
+      if: inputs.use-private-ecr-registry == 'true'
+      uses: aws-actions/amazon-ecr-login@062b18b96a7aff071d4dc91bc00c4c1a7945b076 # v2.0.1
+      env:
+        AWS_REGION: ${{ inputs.ecr-private-aws-region || inputs.aws-region }}
+      with:
+        registries: ${{ inputs.ecr-private-registry }}
+
+    - name: Pull from custom proxy image (private ecr only)
+      if: inputs.use-private-ecr-registry == 'true'
+      shell: bash
+      run: |
+        docker pull ${{ inputs.ecr-private-registry }}.dkr.ecr.${{ inputs.ecr-private-aws-region || inputs.aws-region }}.amazonaws.com/aws-sigv4-proxy:${{ inputs.ecr-private-image-tag }}
+
+    - name: Run aws sig4 proxy container (private ecr only)
+      if: inputs.use-private-ecr-registry == 'true'
+      shell: bash
+      env:
+        AWS_ACCESS_KEY_ID: ${{ env.AWS_ACCESS_KEY_ID }}
+        AWS_SECRET_ACCESS_KEY: ${{ env.AWS_SECRET_ACCESS_KEY }}
+        AWS_SESSION_TOKEN: ${{ env.AWS_SESSION_TOKEN }}
+      run: |
+        docker run --rm -d \
+          -e AWS_ACCESS_KEY_ID \
+          -e AWS_SECRET_ACCESS_KEY \
+          -e AWS_SESSION_TOKEN \
+          -p ${{ inputs.proxy-port }}:${{ inputs.proxy-port }} \
+          -v /tmp/setup-gap:/tls \
+          ${{ inputs.ecr-private-registry }}.dkr.ecr.${{ inputs.ecr-private-aws-region || inputs.aws-region }}.amazonaws.com/aws-sigv4-proxy:${{ inputs.ecr-private-image-tag }} -v \
```

```suggestion
          ${{ inputs.ecr-private-registry }}.dkr.ecr.${{ inputs.ecr-private-aws-region || inputs.aws-region }}.amazonaws.com/aws-sigv4-proxy:${{ inputs.ecr-private-image-tag }} \
```

Does the `-v` flag here increase verbosity on the sigv4 proxy? If so, I don't think we want that, right?

---

I'm not sure how that got there to be honest. Removing.

---

d9224ca426a84e53b45e1a6d92b042e067eb55fb